crates/common/trie/db.rs

```diff
@@ -20,6 +20,11 @@ impl InMemoryTrieDB {
     pub fn new(map: Arc<Mutex<HashMap<Vec<u8>, Vec<u8>>>>) -> Self {
         Self { inner: map }
     }
+    pub fn new_empty() -> Self {
+        Self {
+            inner: Default::default(),
+        }
+    }
```

Comment:
Maybe implement `Default` instead?

Reply:
Good idea

Reply:
I think I'll keep it this way, adding Default is ok until you have to actually use it for the `Trie`, but since the `Trie` requires a `<dyn TrieDB>` you then have to restrict the implemeting type to have `Default`, which does not seems right to me for another databases, for simplicity's sake, I'd leave it like this

Reply:
If you're going to require they have a way to construct an empty value either way, are we not in practice asking them to implement `Default`? And in that case, we can just make it a trait bound for `TrieDB`.